wagtail/snippets/views/snippets.py

```diff
@@ -174,17 +171,25 @@ def get_base_queryset(self):
             self.queryset = self.queryset(self.request)
         return super().get_base_queryset()
 
-    def _get_title_column(self, field_name, column_class=SnippetTitleColumn, **kwargs):
-        # Use SnippetTitleColumn class to use custom template
-        # so that buttons from snippet_listing_buttons hook can be rendered
-        return super()._get_title_column(field_name, column_class, **kwargs)
-
     def get_columns(self):
         return [
             BulkActionsCheckboxColumn("checkbox", accessor=lambda obj: obj),
             *super().get_columns(),
         ]
 
+    def get_list_dropdown_buttons(self, instance, parent_context):
+        buttons = super().get_list_dropdown_buttons(instance, parent_context)
+        next_url = self.request.path
+        button_hooks = hooks.get_hooks("register_snippet_listing_buttons")
```

Comment:
I briefly mentioned this in the PR description,

> As we don't explicitly document nor test custom snippets listing buttons with dropdowns, I opted to move all snippets listing buttons (including ones that come from the hook) to the three-dot dropdown menu, so that it's more consistent with the rest of our universal listings styles.

Though you're correct, it is possible that people have used a `ButtonWithDropdown`/`ButtonWithDropdownFromHook` in here, even if it's not explicitly documented.

> I'd suggest introducing a new `register_snippet_listing_more_buttons` hook here, which will also make it consistent with the page listing. (Except the page listing doesn't currently have corresponding 'construct' hooks, so maybe they need to be added too...)

I thought another important reason why we're doing this is so that we have a way to customise the buttons that's more natural with the other customisations (i.e. via `ViewSet`), making the hooks in this case just kept for "legacy" because (I might've misremembered this) view-level hooks are more suited for code that can alter the response? :D

How about... adding a check, in the buttons returned by the hook, instances of `BaseDropdownButton` are put in the top-level, while others are put inside the default `ButtonWithDropdown` instance?

If that sounds a bit too smart, I'm happy to add `register_snippet_listing_more_buttons`, but as you said, we may have to add the `construct` hooks to the page listing view.

Reply:
Just realised, there's no way that `ButtonWithDropdownFromHook` is used in this hook, because that class requires `page` and `page_perms` in its `__init__`. We only document `ButtonWithDropdownFromHook` and not `ButtonWithDropdown`:

https://docs.wagtail.org/en/stable/reference/hooks.html#buttons-with-dropdown-lists

I think it's safe to say that we never really supported custom dropdown buttons on snippets, but I added 82b305daa1783d8e3d70af53c7c7332cd653e392 anyway in case people use `ButtonWithDropdown`. So what happens now is that instances of `BaseDropdownMenuButton` will be put at the top-level, while other buttons will be put into the default "more" dropdown button.

Technically people can still add top-level non-dropdown buttons using the `construct` hook (and in fact we do have one in the tests).

I've added more tests to ensure this is the case.

Reply:
For the record, https://torchbox.slack.com/archives/C0249LXUTFG/p1694681091208729 is the conversation where @benenright and I settled on the current behaviour for the page listing buttons. Interestingly, one of the points I made there can be turned on its head for snippets...

> and I suppose any time a developer specifically chose to add their button to the main set rather than inside "More", there was probably a good reason for that, so it feels a bit presumptious for us to go "we're going to move it into the More menu, because we want a cleaner design and we know better than you" :slightly_smiling_face:

For snippets, there is no "more_buttons" hook, so that choice wasn't available to developers. And in fact, it appears that the only public code on Github using `register_snippet_listing_buttons` is wagtail-localise and wagtail.contrib.simple_translation - both of which use the "more_buttons" hook for the corresponding actions on the page listing, so it's reasonable to infer that they'd happily do the same for snippets if the option was there. For me, that swings the decision in favour of changing `register_snippet_listing_buttons` to place the buttons into the 'more' dropdown (i.e. your current approach). I'm not sure whether the inconsistency between this and the page listing will become a headache in future, but we can cross that bridge when we come to it... as you point out, we may well end up phasing out these hooks soon in favour of configuring these buttons in the viewset.

Reply:
Thanks, yep I agree. I've added ded26f0d4452aea2bf98eb2d6a632d0843771d5c which passes `more_buttons` (instead of `list_buttons`) to the construct hook.